xds/internal/server/rds_handler.go

```diff
@@ -89,74 +93,93 @@ func (rh *rdsHandler) updateRouteNamesToWatch(routeNamesToWatch map[string]bool)
 			delete(rh.updates, routeName)
 		}
 	}
+}
 
-	// If the full list (determined by length) of updates are now successfully
-	// updated, the listener is ready to be updated.
-	if len(rh.updates) == len(rh.cancels) && len(routeNamesToWatch) != 0 {
-		drainAndPush(rh.updateChannel, rdsHandlerUpdate{updates: rh.updates})
-	}
+// determines if all dynamic RDS needed has received configuration or update.
+func (rh *rdsHandler) determineRDSReady() bool {
+	return len(rh.updates) == len(rh.cancels)
```

Comment:
Hmm yeah, this is safe because only gets written to in the previous instructions in the context of an xDS Client callback. Let me think about how to document this.

Reply:
Added comment "Safe to read cancels because only written to in other parts of xDS Client Callbacks, which are sync."